ui/app/clinical/consultation/directives/consultationConfirmOnExit.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+angular.module('bahmni.clinical')
+    .directive('consultationConfirmOnExit', ['clinicalAppConfigService', 'messagingService', 'spinner', '$state',
+        function (clinicalAppConfigService, messagingService, spinner, $state) {
+            return {
+                link: function ($scope, elem, attrs) {
+                    window.onbeforeunload = function () {
+                        console.log("change");
+                    };
+                    $scope.$on('$stateChangeStart', function (event, next, current) {
+                        var noOfMedications = 0;
+                        var navigating = next.url.split("/")[1];
+                        var allConsultationBoards = clinicalAppConfigService.getAllConsultationBoards();
+                        var outOfConsultationBoard = true;
+                        if ($scope.consultation) {
+                            if ($scope.consultation.orders) {
+                                if ($scope.consultation.orders.length !== $scope.consultation.investigations.length) {
+                                    $state.dirtyConsultationForm = true;
```

Comment:
This can be rewritten as--
`$state.dirtyConsultationForm = $scope.consultation.orders.length !== $scope.consultation.investigations.length`

Reply:
Yes need to assign it true only if the condition is matched. If the previous value is already true, I don't that to get changed, as it might refer to a dirty check of some other tab.